contracts/modules/assessment/Assessment.sol

```diff
@@ -132,25 +132,49 @@ contract Assessment is IAssessment, MasterAwareV2 {
   /// @param to      The member address where the NXM is transfered to. Useful for switching
   ///                membership during stake lockup period and thus allowing the user to withdraw
   ///                their staked amount to the new address when possible.
-  function unstake(uint96 amount, address to) external whenNotPaused override {
+  function unstake(uint96 amount, address to) external whenNotPaused {
+    _unstake(msg.sender, amount, to);
+  }
+
+  /// Withdraws a portion or all of the user's stake
+  ///
+  /// @dev At least stakeLockupPeriodInDays must have passed since the last vote.
+  ///
+  /// @param staker  The address of the staker whose stake will be unstaked
+  /// @param amount  The amount of nxm to unstake
+  /// @param to      The member address where the NXM is transfered to. Useful for switching
+  ///                membership during stake lockup period and thus allowing the user to withdraw
+  ///                their staked amount to the new address when possible.
+  function unstakeFor(address staker, uint96 amount, address to) external whenNotPaused {
+    _unstake(staker, amount, to);
+  }
+
+  function _unstake(address staker, uint96 amount, address to) internal {
+
+    uint stakeAmount = stakeOf[staker].amount;
+    if (amount > stakeAmount) {
+      revert InvalidAmount(stakeAmount);
+    }
```

Comment:
I've add this new validation. If the staker has 0 stake / amount > stake it will just revert with 
```
math underflow check: panic code 0x11
```
which does not seem ideal as it does not provide clarity what went wrong.